dev/devicelab/bin/tasks/gradle_plugin_test.dart

```diff
@@ -46,6 +47,216 @@ Future<void> main() async {
     print('\nUsing JAVA_HOME=$javaHome');
 
     try {
+      await runPluginProjectTest((FlutterPluginProject pluginProject) async {
+        section('APK content for task assembleDebug without explicit target platform');
+        await pluginProject.runGradleTask('assembleDebug');
+
+        if (!pluginProject.hasDebugApk)
+          throw TaskResult.failure(
+              'Gradle did not produce a debug apk file at: ${pluginProject.debugApkPath}');
+
+        final Set<String> apkFiles = await pluginProject.getFilesInApk(pluginProject.debugApkPath);
+
+        _checkItContains<String>(<String>[
+          'AndroidManifest.xml',
+          'classes.dex',
+          'assets/flutter_assets/isolate_snapshot_data',
+          'assets/flutter_assets/kernel_blob.bin',
+          'assets/flutter_assets/vm_snapshot_data',
+          'lib/arm64-v8a/libflutter.so',
+          'lib/armeabi-v7a/libflutter.so',
+          'lib/x86/libflutter.so',
+          'lib/x86_64/libflutter.so',
+        ], apkFiles);
+
+        _checkItDoesNotContain<String>(<String>[
+          'lib/arm64-v8a/libapp.so',
+          'lib/armeabi-v7a/libapp.so',
+          'lib/x86/libapp.so',
+          'lib/x86_64/libapp.so',
+        ], apkFiles);
+      });
+
+      await runPluginProjectTest((FlutterPluginProject pluginProject) async {
+        section('APK content for task assembleDebug with target platform = android-arm');
+        await pluginProject.runGradleTask('assembleDebug',
+            options: <String>['-Ptarget-platform=android-arm']);
+
+        if (!pluginProject.hasDebugApk)
+          throw TaskResult.failure(
+              'Gradle did not produce a debug apk file at: ${pluginProject.debugApkPath}');
+
+        final Set<String> apkFiles = await pluginProject.getFilesInApk(pluginProject.debugApkPath);
+
+        _checkItContains<String>(<String>[
+          'AndroidManifest.xml',
+          'classes.dex',
+          'assets/flutter_assets/isolate_snapshot_data',
+          'assets/flutter_assets/kernel_blob.bin',
+          'assets/flutter_assets/vm_snapshot_data',
+          'lib/armeabi-v7a/libflutter.so',
+          'lib/x86/libflutter.so',
```

Comment:
How come x86 and x86_64 are in here too if we specified `'-Ptarget-platform=android-arm'`?

Reply:
Maybe because it's debug mode, and we proactively include them for simulators?

Reply:
Yup. I added a comment to the test :)